packages/notifier/README.md

```diff
@@ -1,48 +1,59 @@
+For an introduction to these concepts and implementations, 
+see [here](https://javascript.info/async-iterators-generators).
 
+**Important**: In this doc, a *non-final value* is **not** a reference to JavaScript’s `final` keyword. 
+It means a value in a sequence *prior to* the terminal value, i.e., before the successful completion 
+value or the failure reason. The terminal value is also called the *final value*.
+
+For NotifierKit user documentation see 
+[here](https://agoric.com/documentation/distributed-programming.html#notifiers).
+This doc more precisely describes the semantics of the Notifier and the Subscription, their distributed
+system properties, and what this means for when to use each one. 
 
 # Distributed Asynchronous Iteration
 
-JavaScript defines `AsyncGenerator`s, `AsyncIterable`s, and `AsyncIterator`s.
-Each of these manipulate an *async iteration*, an abstract sequence of
-values with the following structure:
-   * Any number of *non-final values* in a fully ordered sequence to be revealed
-     asynchronously over time. These may continue indefinitely or may be
-     terminated in one of two ways:
-   * The async iteration may *finish* successfully, reporting a final
-     *completion value*, or
-   * The async iteration may *fail*, reporting an alleged *reason* for its
-     failure. The reason is typically an error object, and should be, but may
-     in fact be any JavaScript value.
-
-The call `makeNotifierKit()` makes an `{updater, notifier}` pair.
-The call `makeSubscriptionKit()` for makes a similar
-`{publication, subscription}` pair. The first element of each pair (`updater`,
-`publication`) is used to produce the async iteration which can then be
-consumed using the second element of each pair (`notifier`,`subscription`).
-These have much API in common. The `notifier` and `subscription` both implement
-the JavaScript `AsyncIterable` API for consuming the iteration.
-
-There is no JavaScript standard for only producing an iteration, so this
-package defines the `IterationObserver` API. Both `updater` and `publication`
-implement the `IterationObserver` API. For both pairs, the `IterationObserver`
-provides *only* the authority to produce the iteration. The `AsyncIterable`
-provides *only* the authority to consume the iteration.
+An *async iteration* is an abstract sequence of values. It consists of any number of *non-final values* 
+in a fully ordered sequence revealed asynchronously over time. In other words, the values are fully 
+ordered compared to the time when they were provided to the publishing mechanism. In the case of 
+more than one publisher supplying values to the same sequence, the order is determined by when the 
+sequence receives the values. The point is that all consumers see the whole sequence, or a subset of 
+it, in the same order. 
+
+The sequence may continue indefinitely or may terminate in one of two ways:
```

Comment:
Honestly, I think that makes it harder to follow. There are two distinctions going on here. First, non-final vs. final, and second finish vs. failure. Separating discussion/definition of the two makes it clearer and easier to follow. If there'd been a single distinction of non-final vs. final, yeah, put them together. Otherwise it's two separate concepts, so they should each be in their own paragraph or list.

Reply:
I buy that. Thanks.